api/odl.py

```diff
@@ -597,6 +598,45 @@ def _fulfill(
                         content_type = candidate_content_type
                         break
 
+        return content_link, content_type
+
+    def _fulfill(
+        self,
+        loan: Loan,
+        delivery_mechanism: Optional[LicensePoolDeliveryMechanism] = None,
+    ) -> FulfillmentInfo:
+        licensepool = loan.license_pool
+        doc = self.get_license_status_document(loan)
+        status = doc.get("status")
+
+        if status not in [self.READY_STATUS, self.ACTIVE_STATUS]:
+            # This loan isn't available for some reason. It's possible
+            # the distributor revoked it or the patron already returned it
+            # through the DRM system, and we didn't get a notification
+            # from the distributor yet.
+            self.update_loan(loan, doc)
+            raise CannotFulfill()
+
+        expires = doc.get("potential_rights", {}).get("end")
+        expires = dateutil.parser.parse(expires)
+
+        links = doc.get("links", [])
+        content_link, content_type = self._find_content_link_and_type(
+            links, delivery_mechanism
+        )
+
+        if (
+            not content_link
+            and not content_type
+            and licensepool.presentation_edition.medium == Edition.AUDIO_MEDIUM
+        ):
```

Comment:
Now that we are correctly setting the delivery mechanisms, is this condition still needed?

Reply:
I think that this piece of code is still required in the case when `format` is not present in the License Info Document.